### PR TITLE
Fix: Camera movement flagging

### PR DESCRIPTION
**Pull request by fallenoak**
Billboarding needs to update whenever the camera rotation OR position changes, and not just when camera rotation changes.

### CODE REVIEW

**Comment by timkurvers** [comment]
Ah good point! :+1:

What about movement/rotation of the actual billboarded doodad/unit? Not sure if that occurs anywhere in the game.

File: src/components/game/index.jsx [hunk 1]
```diff
@@ -77,15 +78,19 @@ class GameScreen extends React.Component {
...
-    const cameraRotated = this.prevCameraRotation === null ||
-      !this.prevCameraRotation.equals(this.camera.quaternion);
+    const cameraMoved =
+      this.prevCameraRotation === null ||
+      this.prevCameraPosition === null ||
+      !this.prevCameraRotation.equals(this.camera.quaternion) ||
+      !this.prevCameraPosition.equals(this.camera.position);
 
-    session.world.animate(this.clock.getDelta(), this.camera, cameraRotated);
+    session.world.animate(this.clock.getDelta(), this.camera, cameraMoved);
```
Comment (timkurvers) [inline src/components/game/index.jsx]: Don't feel very strongly about this, but should we call this `cameraChanged` instead?

**Comment by fallenoak** [comment]
Hmm, good question. So far, I've only come across billboarding on M2s that are fixed in position (map doodads). Maybe it's best to tackle billboarding for moving M2s when we come across one.

File: src/components/game/index.jsx (hunk 1 above)
Comment (fallenoak) [inline src/components/game/index.jsx]: I guess I viewed the name as reflecting translation + rotation (both are forms of movement). I suppose, in theory, something like the FOV changing could also impact billboarding, so maybe changed would make more sense.

Okay if I do that as a follow up commit? I'm dependent on this commit in another PR I'm about to make.

Comment (timkurvers) [inline src/components/game/index.jsx]: Let's keep it as it is for now :+1: Perhaps it's even possible to mark it on the camera itself – on `userData` perhaps? That would reduce the amount of stuff we're passing around.

Comment (fallenoak) [inline src/components/game/index.jsx]: Good idea!

Let's tackle both the name and locating it within the camera object in a separate PR.

Comment (timkurvers) [inline src/components/game/index.jsx]: :+1:

**Comment by timkurvers** [comment]
Let's do that :+1:

**Comment by timkurvers** [comment]
:tada:

**Comment by fallenoak** [comment]
Woohoo!

**Pull request merged**